spring-mvc-basics-4/src/main/java/com/baeldung/jsonparam/JsonParamController.java

```diff
@@ -0,0 +1,55 @@
+package com.baeldung.jsonparam;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import java.io.IOException;
+
+@Controller
+@RestController
+@RequestMapping("/api")
+public class JsonParamController {
+
+    @GetMapping(value = "/message")
+    public String testQueryParamApi(@RequestParam("jsonParam") String jsonParam) {
```

Comment:
Let's think of a better name for this method. We wouldn't normally start a method in our Controllers with testxxx. Likewise, jsonParam is very vague. What is this param?